cEP-0019.md

```diff
@@ -0,0 +1,467 @@
+   [see sample](https://github.com/coala/gh-board/pull/18#pullrequestreview-116783753)
+
+3. Review Discussion
+   [see sample](https://github.com/coala/gh-board/pull/18#discussion_r186249423)
+
+However, those reactions spread over all PRs and are not collected and
+analyzed. Moreover, GitHub doesn't send notifications to reviewers who receive
+reactions. By tracking reactions, a nice feedback loop would be created
+for both senior and junior reviewers.
+
+## Introduction
+
+The meta-review project is about making use of reactions to build a
+meta-review system, similar to a meta-moderation system. These responses
+are to be collected, processed and displayed on
+[gh-board](https://github.com/coala/gh-board), which is a nice kanban
+board. Compared to [Huboard](https://huboard.com/) which requires server-side
+support, gh-board is serverless.
+
+The scope of this project would exclude non-public meta-reviews. That is,
+there would not be any anonymous meta-reviews as that introduces too many
+additional complexities. The public only version would be used for
+a long time before adding support for non-public meta-reviews.
+
+Note there is a technical limitation in that reactions can not be placed
+onto the "review summary", which means general review comments should not
+be put into the "review summary" text box, as it will be not able to be
+meta-reviewed.
+
+## Objectives
+
+One goal of this project is to encourage people to do meta-reviews,
+systematically. Meta-reviews in need will be displayed on gh-board. The
+implementation details are discussed in the next section.
+
+Also, to encourage people to do more meta-reviews, statistics of meta-reviews
+are to be collected and analysed. People who do meta-reviews will get scores
+according to some metrics. A ranking list is to be displayed on the community
+site. The newcomer guide will also be updated to require newcomers to finish
+at least one meta-review and receive at least one positive meta-review.
+
+To conclude, this project builds a meta-review system on the community site
+and gh-board to display score ranking statistics and meta-reviews in need,
+respectively, and encourage people to do meta-review.
+
+## Meanings of reactions
+
+As mentioned in the background section, GitHub provides 6 types of reactions.
+They have different meanings in the context of meta-review system.
+
+There are 6 types of reactions on GitHub: CONFUSED 😕, HEART ❤️, HOORAY 🎉,
+LAUGH 😄, THUMBS_DOWN 👎, THUMBS_UP 👍.
+
+THUMBS_UP and THUMBS_DOWN are most commonly used. They express strong feelings:
+THUMBS_UP means the review comment is well-written and helpful, while
+THUMBS_DOWN means the review comment is misleading, or even worse, violates
+[coala Community Code of Conduct](https://github.com/coala/cEPs/blob/master/cEP-0006.md).
+They will contribute more to the score which is explained in the
+[Ranking list](#ranking-list) section.
+
+HEART, HOORAY and LAUGH are reactions that express positive feelings, but not
```

Comment:
I guess LAUGH is not able to be easily used for meta-review.

HOORAY on review comments possibly has a few usages, and might need more community input on what meanings it should have, and how it should be part of meta-review.

HEART on review comments possibly has a few usages, but it feels like it is best used for showing a greater amount of appreciation for a review comment than THUMBS_UP.

Reply:
I don't like how GitHub call this emoji 😄. IMO it should not be interpreted as "LAUGH". It's a smiling face, which is a positive feeling, but not as strong as THUMBS_UP. But I think there's an extra benefit if we don't use that in the meta-review system: people know when they use this emoji, it would not be taken account into the meta-review score.

HOORAY sounds like "that's a great idea, let's do it!". For me, it's similar to THUMBS_UP. Maybe even slightly stronger than THUMBS_UP.

HEART is not used very often. Agree, it could be used to show a greater amount of appreciation.

My very first thought was those emojis are similar and their meanings are overlapping. Thus we don't need to give them exact meanings. But it would be great if we clarify them clearly and let everyone in the community comply with it.

Reply:
Due to using gh-board, we can give the user more refined meanings for these reactions.

e.g. providing our meanings as a legend that meta-reviewers can see while doing meta-review.

Re 'LAUGH', even if it is just 'smile' and thus is positive, it will be hard to prevent people from over-using it with less consideration for scoring to express personal jolliness rather than technical appropriateness.  e.g. if a review comment is funny, but not particularly useful, it will attract smile reactions, and possibly lots of them.  This skews the scores, and results in the feedback loop encouraging humour instead of technical reviews.  This reduces the value of any 'smile' meta-reviews which were intended to show appreciation for technically useful review comments.

> Thus we don't need to give them exact meanings.

Maybe instead of "not as strong as THUMBS_UP do. ...", this cEP says "not as clear as THUMBS_UP, and consequently will be given a lower positive score, or excluded from the scoring after analysis of how they are being used."

And again this is leading towards scoring being a per-community thing, with different weightings and even meanings being given by different orgs, and thus it shouldnt be part of gh-board, but should be in the community repo.  The meanings may change over time, which means the changes need to be accompanied with old statistics being archived, and new meta-review using the new scoring system.  That is clearly beyond gh-board's objectives, but it is easy to do with a complex data model which can be modified at any time, in any way, to cater for org needs.

gh-board's UI could perhaps have factual aggregated statistical information like how many of each emoji a person gave/received, again with a legend explaining their intended meaning, but without trying to aggregate across the different reaction types.